provider/pkg/gen/go-templates/apiextensions/customResource.tmpl

```diff
@@ -0,0 +1,131 @@
+	Kind *string `pulumi:"kind"`
+	// Standard object metadata.
+	Metadata *metav1.ObjectMeta `pulumi:"metadata"`
+}
+
+type CustomResourceState struct {
+	// APIVersion defines the versioned schema of this representation of an object. Servers should convert recognized schemas to the latest internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources
+	ApiVersion pulumi.StringPtrInput
+	// Kind is a string value representing the REST resource this object represents. Servers may infer this from the endpoint the client submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+	Kind pulumi.StringPtrInput
+	// Standard object metadata.
+	Metadata metav1.ObjectMetaPtrInput
+}
+
+func (CustomResourceState) ElementType() reflect.Type {
+	return reflect.TypeOf((*customResourceState)(nil)).Elem()
+}
+
+type customResourceArgs struct {
+	// APIVersion defines the versioned schema of this representation of an object. Servers should convert recognized schemas to the latest internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources
+	ApiVersion string `pulumi:"apiVersion"`
+	// Kind is a string value representing the REST resource this object represents. Servers may infer this from the endpoint the client submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+	Kind string `pulumi:"kind"`
+	// Standard object metadata.
+	Metadata *metav1.ObjectMeta `pulumi:"metadata"`
+	// Untyped map that holds any user-defined fields.
+	OtherFields map[string]interface{} `pulumi:""`
+}
+
+// The set of arguments for constructing a Deployment resource.
+type CustomResourceArgs struct {
+	// APIVersion defines the versioned schema of this representation of an object. Servers should convert recognized schemas to the latest internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources
+	ApiVersion pulumi.StringInput
+	// Kind is a string value representing the REST resource this object represents. Servers may infer this from the endpoint the client submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+	Kind pulumi.StringInput
+	// Standard object metadata.
+	Metadata metav1.ObjectMetaPtrInput
+	// Untyped map that holds any user-defined fields.
+	// TODO: do we need to support Inputs in the map?
```

Comment:
I suspect we do - and I think that is possible with the `UntypedArgs` support perhaps?

Reply:
I’ll test that

Reply:
Yep, it works as expected.